src/state.ml

```diff
@@ -192,8 +193,9 @@ type session = {
 }
 
 let init_session ~my_session_id ?(their_session_id = 0L) ?(compress = false)
-    ?(protocol = `Tcp) ~my_hmac ~their_hmac () =
+    ?(protocol = `Tcp) ?(hmac_algorithm = `SHA1) ~my_hmac ~their_hmac () =
```

Comment:
I wonder, is there negotiation of the `auth`? I.e. if a client has one in its configuration, can the server deselect it and choose a different one (i.e. for PUSH_NOTIFY)?

If the answer is no, why should we keep it in `type session`? We can just use the `Config.get Auth` whenever needed.

Reply:
I find the story in openvpn about cipher (and hmac?) negotiation hard to follow. At the moment I don't remember any negotiation about `auth` - except if the cipher is AEAD where `auth` is ignored.

I added it to `session` because we seem to push that "further" than `config`. I also had in mind that we need to choose SHA256 as `auth` for control packets when using tls-crypt-v2, but I'm not sure how I got this idea and I don't think it is the case.

Reply:
I have refactored the code to not store the hmac algorithm in `type session`.

I have not seen anything that suggests the hmac algorithm can be negotiated - other than if you use an AEAD cipher. We probably need to restructure the code in order to support AEAD. I will not think about that right now though.